src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildAWindowsDesktopProject.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using FluentAssertions;
+using Microsoft.NET.TestFramework;
+using Microsoft.NET.TestFramework.Commands;
+using Xunit;
+using Xunit.Abstractions;
+using Microsoft.NET.TestFramework.Assertions;
+using Microsoft.NET.TestFramework.ProjectConstruction;
+
+namespace Microsoft.NET.Build.Tests
+{
+    public class GivenThatWeWantToBuildAWindowsDesktopProject : SdkTest
+    {
+        public GivenThatWeWantToBuildAWindowsDesktopProject(ITestOutputHelper log) : base(log)
+        {}
+
+        [WindowsOnlyRequiresMSBuildVersionTheory("16.7.0-preview-20310-07")]
+        [InlineData("UseWindowsForms")]
+        [InlineData("UseWPF")]
+        public void It_errors_when_missing_windows_target_platform(string propertyName)
+        {
+            var targetFramework = "net5.0";
+            TestProject testProject = new TestProject()
+            {
+                Name = "MissingTargetPlatform",
+                IsSdkProject = true,
+                TargetFrameworks = targetFramework
+            };
+            testProject.AdditionalProperties[propertyName] = "true";
+            testProject.AdditionalProperties["TargetPlatformIdentifier"] = "custom"; // Make sure we don't get windows implicitly set as the TPI
```

Comment:
If you always set `_EnableDefaultWindowsPlatform` to false, then you can remove this line I think.

Reply:
I think we'll be able to remove it when #12492 is fixed but for now we would infer the target platform to be windows here.